src/compressed_chunk.c

```diff
@@ -300,27 +301,33 @@ static void Compressed_Serialize(Chunk_t *chunk,
-    CompressedChunk *compchunk = (CompressedChunk *)malloc(sizeof(*compchunk));
-
-    compchunk->size = readUnsigned(ctx);
-    compchunk->count = readUnsigned(ctx);
-    compchunk->idx = readUnsigned(ctx);
-    compchunk->baseValue.u = readUnsigned(ctx);
-    compchunk->baseTimestamp = readUnsigned(ctx);
-    compchunk->prevTimestamp = readUnsigned(ctx);
-    compchunk->prevTimestampDelta = (int64_t)readUnsigned(ctx);
-    compchunk->prevValue.u = readUnsigned(ctx);
-    compchunk->prevLeading = readUnsigned(ctx);
-    compchunk->prevTrailing = readUnsigned(ctx);
-
-    size_t len;
-    compchunk->data = (uint64_t *)readStringBuffer(ctx, &len);
-    *chunk = (Chunk_t *)compchunk;
-}
+#define COMPRESSED_DESERIALIZE(chunk, ctx, readUnsigned, readStringBuffer, ...)                    \
+    do {                                                                                           \
+        CompressedChunk *compchunk = (CompressedChunk *)malloc(sizeof(*compchunk));                \
+                                                                                                   \
+        compchunk->size = readUnsigned(ctx, ##__VA_ARGS__);                                        \
+        compchunk->count = readUnsigned(ctx, ##__VA_ARGS__);                                       \
+        compchunk->idx = readUnsigned(ctx, ##__VA_ARGS__);                                         \
+        compchunk->baseValue.u = readUnsigned(ctx, ##__VA_ARGS__);                                 \
+        compchunk->baseTimestamp = readUnsigned(ctx, ##__VA_ARGS__);                               \
+        compchunk->prevTimestamp = readUnsigned(ctx, ##__VA_ARGS__);                               \
+        compchunk->prevTimestampDelta = (int64_t)readUnsigned(ctx, ##__VA_ARGS__);                 \
+        compchunk->prevValue.u = readUnsigned(ctx, ##__VA_ARGS__);                                 \
+        compchunk->prevLeading = readUnsigned(ctx, ##__VA_ARGS__);                                 \
+        compchunk->prevTrailing = readUnsigned(ctx, ##__VA_ARGS__);                                \
+                                                                                                   \
+        size_t len;                                                                                \
+        compchunk->data = (uint64_t *)readStringBuffer(ctx, &len, ##__VA_ARGS__);                  \
+        *chunk = (Chunk_t *)compchunk;                                                             \
+        return TSDB_OK;                                                                            \
+                                                                                                   \
+err:                                                                                               \
+        __attribute__((hot, unused));                                                              \
```

Comment:
@OfirMos don't we prefer `cold` instead of `hot` here?
also why do we need the unused label?

Reply:
Oops, right I forgot to put there cold, it's just helps the branch prediction to know if the label is likely to run or unlikely.

Reply:
The unused attribute is to mark for the compiler that on some "instansiation"  of the macro the label will never reached so it not warn us on compilation time.